config/config.go

```diff
@@ -190,24 +191,35 @@ func (c *Config) Validate() error {
 	return nil
 }
 
-// GetSourceIDs returns the Subnet and Chain IDs of all subnets configured as a source
-func (cfg *Config) GetSourceIDs() ([]ids.ID, []ids.ID, error) {
+// GetSourceIDs returns the Subnet, Chain IDs and Allowed Destination Chain IDs of all subnets configured as a source
+func (cfg *Config) GetSourceIDs() ([]ids.ID, []ids.ID, map[ids.ID]map[ids.ID]bool, error) {
```

Comment:
I would also recommend leaving the signature of `GetSourceIDs` as is, and instead adding a separate function to return the mapping of source chains to allowed destinations.

Reply:
We can call this function directly from `teleporter/message_manager.go` rather than having to pass it from `main`.